Source/Dafny/Compilers/SinglePassCompiler.cs

```diff
@@ -1786,13 +1786,17 @@ protected class NullClassWriter : IClassWriter {
               EmitSetterParameter(sw);
             }
           } else if (member is Function fn) {
-            Contract.Assert(fn.Body != null);
-            var w = classWriter.CreateFunction(IdName(fn), CombineAllTypeArguments(fn), fn.Formals, fn.ResultType, fn.tok, fn.IsStatic, true, fn, true, false);
-            EmitCallToInheritedFunction(fn, w);
+            if (!Attributes.Contains(fn.Attributes, "extern")) {
```

Comment:
Can we merge that with L1788?

Reply:
No: for functions with the `:extern` attribute, we would fall into the Contract.Assert case instead of doing nothing

Reply:
Aren't
```c#
else if (member is Function fn) {
            if (!Attributes.Contains(fn.Attributes, "extern")) {
```
and
```c#
else if (member is Function fn && !Attributes.Contains(fn.Attributes, "extern")) {
```
semantically equivalent?

Reply:
No, not when there's another branch below:

```
if (a) {
  if (b) {
    print "ab";
  }
} else {
  print Boom;
}
```

When `a = true, b = false`, the above exits silently, whereas the following prints `"Boom"`


```
if (a &&b) {
  print "ab";
} else {
  print Boom;
}
```

Reply:
![image](https://user-images.githubusercontent.com/8181630/180460187-9abc8055-d501-41f5-94e2-dd6189aa59f0.png)

Reply:
Oh, I see what you mean now. Sorry!